File: Plugins/AdaptiveCruiseControl/controls.py
from ETS2LA.Controls import ControlEvent
from ETS2LA.Utils.translator import _

enable_disable = ControlEvent(
    "toggle_acc",
    _("Toggle Speed Control"),
    "button",
    description=_("When ACC is running this will toggle the speed control on/off."),
    default="n",
)

increment = ControlEvent(
    "increment_speed",
    _("Increase Driving Speed Offset"),
    "button",
    description=_("This will increase the speed offset in the settings."),
    default="↑",
)

decrement = ControlEvent(
    "decrement_speed",
    _("Decrease Driving Speed Offset"),
    "button",
    description=_("This will decrease the speed offset in the settings."),
    default="↓",
)

File: Plugins/AdaptiveCruiseControl/Controls.cs
using ETS2LA.Shared;

namespace AdaptiveCruiseControl;

public static class Controls
{
    public static ControlBinding ToggleAcc => new(AccSettingKeys.Toggle, "Toggle Speed Control", "button", "Toggle ACC on/off.", "n");
    public static ControlBinding IncreaseOffset => new(AccSettingKeys.IncSpeed, "Increase Speed Offset", "button", "Increase ACC speed offset.", "UpArrow");
    public static ControlBinding DecreaseOffset => new(AccSettingKeys.DecSpeed, "Decrease Speed Offset", "button", "Decrease ACC speed offset.", "DownArrow");
}

public record ControlBinding(string Id, string Title, string Type, string Description, string Default);
